reconcile/external_resources/manager.py

```diff
@@ -73,6 +74,42 @@ def setup_factories(
     return of
 
 
+class ExternalResourceDryRunsValidator:
+    def __init__(
+        self,
+        state_manager: ExternalResourcesStateDynamoDB,
+        er_inventory: ExternalResourcesInventory,
+    ):
+        self.state_mgr = state_manager
+        self.er_inventory = er_inventory
+
+    def _check_output_resource_name_duplications(
+        self,
+    ) -> None:
+        seen = set()
+
+        for spec in self.er_inventory.values():
+            output_key = (
+                spec.cluster_name,
+                spec.namespace_name,
+                spec.output_resource_name,
+            )
+            if output_key in seen:
+                raise ExternalResourceOutputResourceNameDuplication(output_key)
+            seen.add(output_key)
```

Comment:
if you want to find all duplicates, can either use another set to store duplicates or use `Counter`

```python
specs = Counter(
  (
     spec.cluster_name,
     spec.namespace_name,
     spec.output_resource_name,
  )
  for spec in self.er_inventory.values()
)

duplicates = [key for key, count in specs.items() if count > 1]
```

Reply:
I prefer to fail fast at pr-checks, there is no need to go through all the objects here as the ones duplicating will be the ones in the related PR.

Reply:
Implemented this way.